src/sodiumCompatibility/java/net/coderbot/iris/compat/sodium/impl/vertex_format/entity_xhfp/QuadViewEntity.java

```diff
@@ -0,0 +1,55 @@
+package net.coderbot.iris.compat.sodium.impl.vertex_format.entity_xhfp;
+
+import net.coderbot.iris.compat.sodium.impl.vertex_format.QuadView;
+import net.coderbot.iris.vertices.NormalHelper;
+import org.lwjgl.system.MemoryUtil;
+
+import java.nio.ByteBuffer;
+
+public class QuadViewEntity extends QuadView {
+	private ByteBuffer buffer;
+	private long writePointer;
+	private int stride = 48;
+	private boolean unsafe;
+
+	public void setup(ByteBuffer buffer, int writeOffset, int stride) {
+		this.buffer = buffer;
+		this.writePointer = writeOffset;
+		this.stride = stride;
+		this.unsafe = false;
+	}
+
+	public void setup(long writePointer, int stride) {
+		this.writePointer = writePointer;
+		this.stride = stride;
+		this.unsafe = true;
+	}
+
+	public float x(int index) {
+		return getFloat(writePointer - stride * (4L - index));
+	}
+
+	public float y(int index) {
+		return getFloat(writePointer + 4 - stride * (4L - index));
+	}
+
+	public float z(int index) {
+		return getFloat(writePointer + 8 - stride * (4L - index));
+	}
+
+	public float u(int index) {
+		return getFloat(writePointer + 16 - stride * (4L - index));
+	}
+
+	public float v(int index) {
+		return getFloat(writePointer + 20 - stride * (4L - index));
+	}
+
+	private float getFloat(long writePointer) {
+		if (unsafe) {
+			return MemoryUtil.memGetFloat(writePointer);
+		} else {
+			return buffer.getFloat((int) writePointer);
+		}
+	}
```

Comment:
I'm not sure if putting this behind a branch is the best approach for performance, maybe we should just have a subclass? I guess we could check the generated assembly to see if this is getting optimized out or not though